presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -286,11 +289,103 @@ public void testEnum()
+        // minutes offset change since 1970-01-01, no DST
+        // using 2 non-JVM zones so that we don't need to worry what Postgres system zone is
+        ZoneId kathmandu = ZoneId.of("Asia/Kathmandu");
+
+        LocalDateTime timeGapInKathmandu = LocalDateTime.of(1986, 1, 1, 0, 13, 7);
+        checkIsGap(kathmandu, timeGapInKathmandu);
+
+        for (ZoneId sessionZone : ImmutableList.of(/*ZoneOffset.UTC,*/ jvmZone/*, vilnius, kathmandu, ZoneId.of(TestingSession.DEFAULT_TIME_ZONE_KEY.getId())*/)) {
+            DataTypeTest tests = DataTypeTest.create()
+                    .addRoundTrip(timestampDataType(), beforeEpoch)
+                    .addRoundTrip(timestampDataType(), afterEpoch)
+                    .addRoundTrip(timestampDataType(), timeDoubledInJvmZone)
+                    .addRoundTrip(timestampDataType(), timeDoubledInVilnius);
+
+            if (!insertWithPresto) {
+                // when writing Postgres JDBC driver converts LocalDateTime to string representing date-time in JVM zone
+                // TODO upgrade drive of find a different way to write timestamp values
+                addTimestampTestIfSupported(tests, legacyTimestamp, sessionZone, epoch); // epoch also is a gap in JVM zone
+                addTimestampTestIfSupported(tests, legacyTimestamp, sessionZone, timeGapInJvmZone);
+                addTimestampTestIfSupported(tests, legacyTimestamp, sessionZone, timeGapInJvmZone2);
+            }
+
+            addTimestampTestIfSupported(tests, legacyTimestamp, sessionZone, timeGapInVilnius);
+            addTimestampTestIfSupported(tests, legacyTimestamp, sessionZone, timeGapInKathmandu);
+
+            Session session = Session.builder(getQueryRunner().getDefaultSession())
+                    .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(sessionZone.getId()))
+                    .setSystemProperty("legacy_timestamp", Boolean.toString(legacyTimestamp))
+                    .build();
+
+            if (insertWithPresto) {
+                tests.execute(getQueryRunner(), session, prestoCreateAsSelect(session, "test_timestamp"));
+            }
+            else {
+                tests.execute(getQueryRunner(), session, postgresCreateAndInsert("tpch.test_timestamp"));
+            }
+        }
+    }
+
+    private void addTimestampTestIfSupported(DataTypeTest tests, boolean legacyTimestamp, ZoneId sessionZone, LocalDateTime dateTime)
+    {
+        if (legacyTimestamp && isGap(sessionZone, dateTime)) {
+            // in legacy timestamp semantics we cannot represent this dateTime
+            return;
+        }
+
+        tests.addRoundTrip(timestampDataType(), dateTime);
+    }
+
+    @DataProvider
+    public Object[][] testTimestampDataProvider()
+    {
+        return new Object[][] {
+                {true, true},
+                {false, true},
+                {true, false},
+                {false, false},
+        };
+    }
+
+    private static void checkIsGap(ZoneId zone, LocalDateTime dateTime)
```

Comment:
How about `assertGap` and change the below to use `assertTrue`

Reply:
This is not tests' assertion (i.e. not something the test is supposed to test). This is verification of test setup, so `check` is more suitable.